src/messages.rs

```diff
@@ -68,6 +73,62 @@ impl From<Message> for Vec<u8> {
     }
 }
 
+impl TryFrom<(libp2p::gossipsub::Message, pchain_types::cryptography::PublicAddress)> for Message {
+    type Error = MessageConversionError;
+
+    fn try_from((message , local_public_address): (libp2p::gossipsub::Message, pchain_types::cryptography::PublicAddress)) 
```

Comment:
@AlvinHon, also please advise if this should be in message.rs or in conversion.rs? I'm not sure about this.

Reply:
Move this and the MessageConversionError into conversions.rs as these are all related to conversions